avocado/core/jobdata.py

```diff
@@ -0,0 +1,217 @@
+    """
+    recorded_mux = os.path.join(resultsdir, JOB_DATA_DIR, MUX_FILENAME)
+    if not os.path.exists(recorded_mux):
+        recorded_mux = os.path.join(resultsdir,
+                                    JOB_DATA_FALLBACK_DIR,
+                                    MUX_FILENAME)
+        if not os.path.exists(recorded_mux):
+            return None
+
+    with open(recorded_mux, 'r') as mux_file:
+        return pickle.load(mux_file)
+
+
+def retrieve_args(resultsdir):
+    """
+    Retrieves the job args from the results directory.
+    """
+    recorded_args = os.path.join(resultsdir, JOB_DATA_DIR, ARGS_FILENAME)
+    if not os.path.exists(recorded_args):
+        recorded_args = os.path.join(resultsdir,
+                                     JOB_DATA_FALLBACK_DIR,
+                                     ARGS_FILENAME)
+        if not os.path.exists(recorded_args):
+            return None
+
+    with open(recorded_args, 'r') as args_file:
+        return pickle.load(args_file)
+
+
+def retrieve_config(resultsdir):
+    """
+    Retrieves the job settings from the results directory.
+    """
+    recorded_config = os.path.join(resultsdir, JOB_DATA_DIR, CONFIG_FILENAME)
+    if not os.path.exists(recorded_config):
+        recorded_config = os.path.join(resultsdir,
+                                       JOB_DATA_FALLBACK_DIR,
+                                       CONFIG_FILENAME)
+        if not os.path.exists(recorded_config):
+            return None
+
+    return recorded_config
+
+
+def retrieve_cmdline(resultsdir):
+    """
+    Retrieves the job command line from the results directory.
+    """
+    recorded_cmdline = os.path.join(resultsdir, JOB_DATA_DIR, CMDLINE_FILENAME)
+    if not os.path.exists(recorded_cmdline):
+        recorded_cmdline = os.path.join(resultsdir,
+                                        JOB_DATA_FALLBACK_DIR,
+                                        CMDLINE_FILENAME)
+        return None
+
+    with open(recorded_cmdline, 'r') as cmdline_file:
+        return ast.literal_eval(cmdline_file.read())
+
+
+def get_resultsdir(logdir, jobid):
+    """
+    Gets the job results directory using a Job ID.
+    """
+    if jobid == 'latest':
+        try:
+            actual_dir = os.readlink(os.path.join(logdir, 'latest'))
+            return os.path.join(logdir, actual_dir)
+        except IOError:
+            return None
+
+    matches = 0
+    short_jobid = jobid[:7]
+    if len(short_jobid) < 7:
+        short_jobid += '*'
+    idfile_pattern = os.path.join(logdir, 'job-*-%s' % short_jobid, 'id')
+    for id_file in glob.glob(idfile_pattern):
+        if get_id(id_file, jobid) is not None:
+            match_file = id_file
+            matches += 1
+            if matches > 1:
+                from logging import getLogger
+                getLogger("avocado.app").error("hash '%s' is not unique "
+                                               "enough", jobid)
+                sys.exit(exit_codes.AVOCADO_JOB_FAIL)
```

Comment:
We should not `sys.exit` directly from inside the library. It should raise an exception and the caller should decide what to do.